MIGRATION.md

```diff
@@ -0,0 +1,76 @@
+# Migrate PDP storage to Manage
+
+The PDP policies used to be stored by the PDP in its own database and administered in its own GUI.
+This is deprecated and this guide describes how to migrate your rules management into Manage.
+The PDP remains as the server (with only an API) which decides on the policies at login time.
+
+## Background
+
+Because Manage did not exist yet at the time, the PDP was built with a separate admin GUI and database storage of its policies. Although necessary at the time, this is now inconvenient because Manage does not proviode any information that/which rules exist for an entity, there's two admin UIs to work in which leads to confusion, there's only loose referential checking possible (e.g. renaming an entityID drops any associated policies on the floor), and two UIs need to be maintained codewise.
+
+Now that Manage is there, we can move storage of the policies into the Manage backend (MongoDB) and frontend, so we can make use of the existing facilities for managing the history, integrate the admin UI and enforce referential integrity.
+
+The storage in Manage is done in JSON format. Upon PUSH in Manage this is sent to the PDP-server which translates it to XACML and stores it in its local database for runtime evaluation. The PUSH in Manage is when policy changes become active, no delays anymore.
+
+## How does the migration work
+
+A new policies table is added to PDP next to the existing one. Manage can import the existing policies, process and store them and push them to the new table.
+
+The migration can be done in two stages:
+1. The new table exists besides the old one but is not used for policy evaluation. Manage can pull from the old table (as often as desired), and push back to the new table. Manage can then also show any differences between old and new, so certainty can be gained that the conversion is successful. Changing policies at this time still happens from PDP GUI.
+2. The new table is switched to production. The old table is no longer used and the PDP GUI also not.
```

Comment:
I believe the new table is only used for testing. When you switch off the test mode in the pdp config the rules are pushed to the old (ie the live production) table.

Reply:
Confirmed